### ISSUE TITLE
Test for r.geomorphon fails due to optional with default confusion

### ISSUE THREAD

**Issue by wenzeslaus** [OSGeo/grass#2029]
Test fails with `option <profiledata> requires <coordinates>` because there is confusion about optional parameters with default values. This seems to be more an issue with to pygrass and GUI than with r.geomorphon or parser according to analysis by @infrastation and others.

History (in r.info) and pygrass Module fill in the defaults (so `profiledata="-" profileformat="json" is put there`). However, when such command is executed, it fails with:

```
ERROR: Option <profiledata> requires <coordinates>
ERROR: Option <profileformat> requires <coordinates>
```

This happened only after I suggested to @infrastation to use the automated check declaration instead of a manual checks. For r.geomorphon, the solution is either going back to the manual check or not providing defaults which may actually make sense in this case. However, it seems that the issue is in pygrass and the same issue happens to be in GUI too. Maybe the parser can be modified because the defaults are sometimes recorded even when not used (for history), but it seems to me now that the first issue is that the Python code (both GUI and independently pygrass) is mishandling the defaults.

_Originally posted by @wenzeslaus in <https://github.com/OSGeo/grass/issues/1157#issuecomment-770327569>. See also the following comments there._

**Comment by infrastation** [comment]
It might help to add a few paragraphs to `gislib_cmdline_parsing.html` to define more precisely what "default" and "optional" means; and only then to eliminate discrepancies in the offending plugins and parts of the API.

**Comment by ninsbl** [comment]
Tests run successfully again with #2154 See: https://github.com/OSGeo/grass/runs/5260182873?check_suite_focus=true#step:10:97

**Comment by ninsbl** [comment]
Fixed in #2226
